File: client-java/controller/src/main/java/org/evomaster/client/java/controller/mongo/selectors/AndSelector.java
package org.evomaster.client.java.controller.mongo.selectors;

import org.evomaster.client.java.controller.mongo.operations.*;
import org.evomaster.client.java.controller.mongo.QueryParser;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

/**
 *  Represents a selector for the MongoDB `$and` operator.
 * { $and: [ { expression1 }, { expression2 } , ... , { expressionN } ] }
 */
public class AndSelector extends MultiConditionQuerySelector {

    public static final String AND_OPERATOR = "$and";

    @Override
    protected QueryOperation composeConditions(List<QueryOperation> conditions) {
        Objects.requireNonNull(conditions);
        return conditions.isEmpty() ? null : new AndOperation(conditions);
    }

    @Override
    protected String operator() {
        return AND_OPERATOR;
    }
}

File: client-java/controller/src/main/java/org/evomaster/client/java/controller/mongo/selectors/AndSelector.kt
package org.evomaster.client.java.controller.mongo.selectors

import org.bson.Document
import org.evomaster.client.java.controller.mongo.operations.*
import org.evomaster.client.java.controller.mongo.QueryParser

/**
 * { $and: [ { <expression1> }, { <expression2> } , ... , { <expressionN> } ] }
 */
class AndSelector : MultiConditionQuerySelector() {
    override fun operator(): String = "\$and"

    override fun parseConditions(value: Any?): QueryOperation? {
        return when (value) {
            is List<*> -> {
                val conditions = value.map { condition -> QueryParser().parse( condition as Document)}
                AndOperation(conditions)
            }
            else -> null
        }
    }
}